pkg/internal/cluster/create/actions/kubeadmjoin/join.go

```diff
@@ -167,12 +170,13 @@ func runKubeadmJoinControlPlane(
 		containerPath := path.Join("/etc/kubernetes/pki", fileName)
 		// set the path of the certificate into the tmp area on the host
 		tmpPath := filepath.Join(tmpDir, fileName)
+		// TODO(bentheelder): refactor to copy between nodes instead
 		// copies from bootstrap control plane node to tmp area
-		if err := controlPlaneHandle.CopyFrom(containerPath, tmpPath); err != nil {
+		if err := docker.CopyFrom(controlPlaneHandle.String(), containerPath, tmpPath); err != nil {
```

Comment:
this is a major TODO, however it still works for now.

this and the call below must be replaced before we can implement another provider.

this is the last thorny change. I have a plan for it, but I want to stop increasing the size of this PR and iterate since it is still _functionally correct_ for now, I know it took a while to read (thank you @amwat @neolit123 🙏 )